src/main/java/ru/nmedvedev/service/spendmoneyreminder/SpendMoneyReminderBusinessLogicGateService.java

```diff
@@ -0,0 +1,18 @@
+package ru.nmedvedev.service.spendmoneyreminder;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class SpendMoneyReminderBusinessLogicGateService {
+
+    public boolean needToSendNotification(ReminderDayEnum day, Double amount) {
+        switch (day) {
+            case NOT_A_DAY_FOR_A_REMINDER: return false;
+            case HALF_MONTH: return amount >= 1500;
+            case LAST_WORKING_DAY_MINUS_FOUR: return amount >= 1000;
+            case LAST_WORKING_DAY_MINUS_TREE: return amount >= 500;
+            case LAST_WORKING_DAY_MINUS_ONE: return amount >= 400;
```

Comment:
Maximum balance may change one day so these values should be modified. I suggest you to move the to the app config.